pkg/services/ngalert/eval/eval.go

```diff
@@ -776,6 +770,38 @@ func evaluateExecutionResult(execResults ExecutionResults, ts time.Time) Results
 	return evalResults
 }
 
+func buildResult(f *data.Frame, val *float64, ts time.Time) Result {
+	r := Result{
+		Instance:           f.Fields[0].Labels,
+		EvaluatedAt:        ts,
+		EvaluationDuration: time.Since(ts),
+		EvaluationString:   extractEvalString(f),
+		Values:             extractValues(f),
+	}
+	switch {
+	case val == nil:
+		r.State = NoData
+	case *val == 0:
+		r.State = Normal
+	default:
+		r.State = Alerting
+	}
+	return r
+}
+
+func isScalarInstantVector(f *data.Frame) bool {
```

Comment:
See above, something like this (other changes needed).

```suggestion
func scalarInstantVector(f *data.Frame) (float64, bool) {
```